app/src/main/kotlin/com/wire/android/ui/authentication/login/LoginViewModel.kt

```diff
@@ -1,20 +1,84 @@
 package com.wire.android.ui.authentication.login
 
+import androidx.annotation.VisibleForTesting
 import androidx.compose.material.ExperimentalMaterialApi
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
+import com.wire.android.di.ClientScopeProvider
+import com.wire.android.navigation.BackStackMode
+import com.wire.android.navigation.NavigationCommand
+import com.wire.android.navigation.NavigationItem
 import com.wire.android.navigation.NavigationManager
+import com.wire.kalium.logic.data.client.ClientCapability
+import com.wire.kalium.logic.feature.client.RegisterClientResult
 import dagger.hilt.android.lifecycle.HiltViewModel
 import kotlinx.coroutines.launch
 import javax.inject.Inject
+import com.wire.kalium.logic.data.user.UserId
+import com.wire.kalium.logic.feature.auth.AddAuthenticatedUserUseCase
+import com.wire.kalium.logic.feature.auth.AuthenticationResult
 
 @ExperimentalMaterialApi
 @HiltViewModel
-class LoginViewModel @Inject constructor(private val navigationManager: NavigationManager) : ViewModel() {
+open class LoginViewModel @Inject constructor(
```

Comment:
I agree that it should be refactored just maybe it would be better to extract it to another ViewModel and not combine it with `LoginViewModel` because this one is supposed to work with the `LoginScreen` and there is no need for all these methods, it's only purpose is to handle the navigation but but that may just be my opinion, I'm open to counter-arguments. 😄